pkg/operator/ceph/disruption/machinelabel/reconcile.go

```diff
@@ -0,0 +1,145 @@
+	MachineFencingLabelKey          = "fencegroup.rook.io/cluster"
+	MachineFencingNamespaceLabelKey = "fencegroup.rook.io/clusterNamespace"
+	OSDPodLabelValue                = "rook-ceph-osd"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+type ReconcileMachineLabel struct {
+	scheme  *runtime.Scheme
+	client  client.Client
+	options *controllerconfig.Context
+}
+
+type machine struct {
+	isOccupiedByOSD bool
+	RawMachine      mapiv1.Machine
+}
+
+// Reconcile is the implementation of reconcile function for ReconcileMachineLabel
+// which ensures that the machineLabel for the osd pods are in correct state
+func (r *ReconcileMachineLabel) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger.Debugf("reconciling %s", request.NamespacedName)
+
+	// Fetch list of osd pods for the requested ceph cluster
+	pods := &corev1.PodList{}
+	err := r.client.List(context.TODO(), pods, client.InNamespace(request.Namespace), client.MatchingLabels{k8sutil.AppAttr: OSDPodLabelValue, k8sutil.ClusterAttr: request.Name})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Fetching the cephCluster
+	cephClusterInstance := &cephv1.CephCluster{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, cephClusterInstance)
+	if errors.IsNotFound(err) {
+		logger.Infof("cephCluster instance not found for %s", request.NamespacedName)
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		logger.Errorf("failed to fetch cephCluster %s: %+v", request.NamespacedName, err)
+		return reconcile.Result{}, err
+	}
+
+	// skipping the reconcile since the feature is switched off
+	if !cephClusterInstance.Spec.DisruptionManagement.ManageMachineDisruptionBudgets {
+		logger.Debugf("Skipping reconcile for cephCluster %s as manageMachineDisruption is turned off", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch list of machines available
+	machines := &mapiv1.MachineList{}
+	err = r.client.List(context.TODO(), machines, client.InNamespace(cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace))
+	if err != nil {
+		logger.Errorf("failed tp fetch machine list %+v", machines)
+		return reconcile.Result{}, err
+	}
+
+	nodeMachineMap := map[string]machine{}
+
+	// Adding machines to nodeMachineMap
+	for _, m := range machines.Items {
+		if m.Status.NodeRef != nil {
+			nodeMachineMap[m.Status.NodeRef.Name] = machine{RawMachine: m}
+		}
+	}
+
+	// Marking machines that are occupied by the osd pods
+	for _, pod := range pods.Items {
+		if pod.Spec.NodeName != "" {
+			if machine, p := nodeMachineMap[pod.Spec.NodeName]; p {
+				machine.isOccupiedByOSD = true
+				nodeMachineMap[pod.Spec.NodeName] = machine
+			}
+		}
+	}
+
+	// Updating the machine status
+	for _, machine := range nodeMachineMap {
+		labels := machine.RawMachine.GetLabels()
+		clusterName, isClusterNamePresent := labels[MachineFencingLabelKey]
```

Comment:
These values are only used by two nearly identical and fairly long if conditions. Why not write a function that will test if clusterName and clusterNamespace equal expected values that automatically also check the `isClusterNamePresent` and `isClusterNamespacePresent`?

Reply:
Agreed, will simplify